sdk/servicebus/Azure.Messaging.ServiceBus/src/Addons/ServiceBusClientExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Azure.Messaging.ServiceBus.Addons
+{
+    /// <summary>
+    /// These extensions allow additional functionality for the <see cref="ServiceBusClient"/>.
+    /// </summary>
+    public static class ServiceBusClientExtensions
```

Comment:
Should this be ServiceBusSenderExtensions?

Reply:
Is the ServiceBusSender the only place you create a message (meaning, the only consumer of messages)? Also, that means it requires creating a sender to make a message. I did it on the top-level client, because that is (likely) the earliest thing made. However, since that client is disposable, it might make sense to do it on the sender.

Reply:
You can only send a message from the sender. Nothing stops you from creating it in other contexts, but you need a sender to send it. The receiver/processor gets back ServiceBusReceivedMessage which is not directly creatable by the user.

Reply:
I don't really have a strong opinion either way. I think the way you have it makes sense.

Reply:
I think we'll end up with a set of extensions for the sender and receiver, at the end of the day.   I'm a fan of keeping those scoped to an extensions class that matches the target type, personally.

I don't have a good idea of what we'd do in the receiver case, maybe I'm overthinking.

Reply:
One other thought - since these aren't actually extension methods, I suppose we can't rely on them being hidden from Intellisense.

Reply:
ooohhh.... I missed that.  Yeah, I'm guessing that was a mistake and Yanni meant to take the client instance for an extension method.   If not, I'd ask that we consider doing so.

Reply:
Oh right. That is my mistake :-P

Reply:
This has been fixed now. And, it is correctly on the `ServiceBusSender`.